### PR TITLE
Feature/ui adapter prototype integration 4

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds the ability to save an adapter as draft.
- Adds the ability to draw the DAG from drafts.
- General code cleanup

###### NOTE: 
 - The error in properties modal is fixed.
 - The Name of the adapter when it is created is fixed to be shown (instead of just showing 'successfully published' message)

##### TODO:
- Remember the position of nodes while creating a draft 

adaptercreate

### CODE REVIEW

File: cdap-ui/app/features/adapters/controllers/create/canvas-ctrl.js [hunk 1]
```diff
@@ -146,5 +144,35 @@ angular.module(PKG.name + '.feature.adapters')
...
 
+    function setNodesAndConnectionsFromDraft(data) {
+      var ui = data.ui;
+      var config = data.config;
+      var nodes;
+      // Purely for feeding my-plumb to draw the diagram
+      // if I already have the nodes and connections
+      if (ui && ui.nodes) {
+        nodes = ui.nodes;
+        Object.keys(nodes).forEach(function(node) {
```
Comment (elfenheart) [inline cdap-ui/app/features/adapters/controllers/create/canvas-ctrl.js]: you can do angular.forEach(nodes, function (value, key) { ... }) here

Comment (ajainarayanan) [inline cdap-ui/app/features/adapters/controllers/create/canvas-ctrl.js]: True. Thought this was simpler (native). Will change it.

Comment (ajainarayanan) [inline cdap-ui/app/features/adapters/controllers/create/canvas-ctrl.js]: Fixed.

**Comment by elfenheart** [comment]
:+1: LGTM

**Comment by ajainarayanan** [comment]
Merged https://github.com/caskdata/cdap/pull/3197 PR and added animation to plugin properties modal. Since the previous was reviewed I am going to merge this PR since it already has a LGTM

**Pull request merged**